src/terminal/char_grid/cursor.rs

```diff
@@ -24,10 +24,20 @@ pub struct Cursor {
     pub coords: Coords,
     pub style: Styles,
     pub text_style: Styles,
+    config: Config,
```

Comment:
This will raise lifetime issues. Ultimately `Cursor` and `CharGrid` and so on are all owned `Terminal`, which also owns the `Config`. You can't store a reference into the same object.